docs/spec.md

```diff
@@ -2633,6 +2633,52 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills an `output` tensor with values in increasing order starting from zero
+along the `iota_dimension` dimension. `output` tensor shape is specified
+dynamically via `output_shape` input. More formally,
+
+`output[result_index] = constant(is_quantized(output) ?
+quantize(result_index[iota_dimension], element_type(output)) :
+result_index[iota_dimension], element_type(output))`.
+
+#### Inputs
+
+| Label | Name             | Type                                                                               | Constraints |
+|-------|------------------|------------------------------------------------------------------------------------|-------------|
+| (I1)  | `iota_dimension` | `si64`                                                                             | (C1)        |
+| (I2)  | `output_shape`   | 1-dimensional tensor constant of type `si64`                                       | (C1), (C2)  |
+
+#### Outputs
+
+| Name     | Type                                                                             | Constraints |
+|----------|----------------------------------------------------------------------------------|-------------|
+| `output` | tensor of integer, floating-point or complex type or per-tensor quantized tensor | (C2)        |
+
+#### Constraints
+
+* (C1) `0 <= iota_dimension < size(output_shape)`.
+* (C2) `shape(output) = output_shape`.
+
+#### Examples
+
+```mlir
+
+// %output_shape = [4, 5]
+%output = "stablehlo.dynamic_iota"(%output_shape) {
+  iota_dimension = 0 : i64
+} : (tensor<2xi64>) -> tensor<?x?xi32>
```

Comment:
`} : (tensor<2xi64>) -> tensor<?x?xi64>`

Reply:
Is this for consistency, or some other reason? The example for `iota` uses `i32` as the result's element type. I kinda like using different types because it emphasizes that the type of the dimension sizes is unrelated to the element type of the output tensor.

Reply:
Yes for consistency and preferring to test highest bitwidth per [G1.2](https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines)

Reply:
tensor element type mismatch is expected between `output_shape` and `output`

I kept it mismatched for the same reason.

Reply:
missed previous reply. Thanks. 

will change it to highest bitwidth (i64) as per [G1.2](https://github.com/openxla/stablehlo/blob/main/docs/reference.md#testing-guidelines)

Reply:
Spec does not constrain this, so it should be clear there can be mismatches. I think this is generally the case too; I can't think of an op at the top of my head that requires attribute element type to match result element type.